pandas/core/tools/datetimes.py

```diff
@@ -68,13 +68,13 @@
 
     from pandas import Series
 
-# ---------------------------------------------------------------------
-# types used in annotations
+    # ---------------------------------------------------------------------
+    # types used in annotations
 
-ArrayConvertible = Union[List, Tuple, AnyArrayLike, "Series"]
-Scalar = Union[int, float, str]
-DatetimeScalar = TypeVar("DatetimeScalar", Scalar, datetime)
-DatetimeScalarOrArrayConvertible = Union[DatetimeScalar, ArrayConvertible]
+    ArrayConvertible = Union[List, Tuple, AnyArrayLike, Series]
+    Scalar = Union[int, float, str]
+    DatetimeScalar = TypeVar("DatetimeScalar", Scalar, datetime)
+    DatetimeScalarOrArrayConvertible = Union[DatetimeScalar, ArrayConvertible]
```

Comment:
I guess this is also an option. would prefer to update regex though.

Reply:
Sure - this wasn't for the regex though, but rather because otherwise we get
```
ImportError while loading conftest '/home/marco/pandas-marco/pandas/conftest.py'.
pandas/__init__.py:51: in <module>
    from pandas.core.api import (
pandas/core/api.py:31: in <module>
    from pandas.core.groupby import Grouper, NamedAgg
pandas/core/groupby/__init__.py:1: in <module>
    from pandas.core.groupby.generic import DataFrameGroupBy, NamedAgg, SeriesGroupBy
pandas/core/groupby/generic.py:69: in <module>
    from pandas.core.frame import DataFrame
pandas/core/frame.py:156: in <module>
    from pandas.core.series import Series
pandas/core/series.py:108: in <module>
    from pandas.core.tools.datetimes import to_datetime
pandas/core/tools/datetimes.py:74: in <module>
    ArrayConvertible = Union[List, Tuple, AnyArrayLike, Series]
E   NameError: name 'Series' is not defined
```
because here `Series` is only defined if `TYPE_CHECKING`

Reply:
https://www.python.org/dev/peps/pep-0563/#forward-references goes on to say

> Depending on the specific case, some of the cases listed above might be worked around by placing the usage in a if TYPE_CHECKING: block. This will not work for any code that needs to be available at runtime, notably for base classes and casting. For named tuples, using the new class definition syntax introduced in Python 3.6 solves the issue.

so this is fine